### PR TITLE
PLAT-128895: Fixed transition direction for meridiem/day when hour/month changes

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
Enact-DCO-1.0-Signed-off-by: Daniel Stoian daniel.stoian@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- applied `agate/internal/Picker/ChangeAdapter` to `agate/Picker` and `agate/RangePicker` instead of `agate/internal/Picker` in order to have the `reverseTransition` prop available in `agate/internal/DateTime/DateTimeDecorator`
- changed usage of `RangePicker` with `RangePickerBase` in `internal/DateComponentPicker/DateComponentRangePicker.js` in order to have the `reverseTransition` information available.
- set the `reverseTransition` props in `DatePickerBase` based on the state from `DateTimeDecorator`
- set the `reverseTransition` prop  for meridiem picker in `DatePickerBase` to null

### Links
[//]: # (Related issues, references)
PLAT-128895

### Comments

### CODE REVIEW

**Review by stanca-pop-lgp — APPROVED** [review]
LGTM

**Review by 0x64 — APPROVED** [review]
:lgtm:

File: DatePicker/DatePicker.js [hunk 1]
```diff
@@ -29,13 +29,14 @@ const getLabelFormatter = () => new DateFmt({
 });
 
 const dateTimeConfig = {
-	customProps: function (i18n, value, props) {
+	customProps: function (i18n, value, reverseTransition,  props) {
```
Comment (MikyungKim) [inline DatePicker/DatePicker.js:32]:
```suggestion
	customProps: function (i18n, value, reverseTransition, props) {
```

Comment (daniel-stoian-lgp) [reply DatePicker/DatePicker.js]: 👍

**Review by MikyungKim — COMMENTED** [review]
LGTM for me, but I need @0x64 's opinion when he's back :)

File: DatePicker/DatePickerBase.js [hunk 2]
```diff
@@ -94,6 +94,14 @@ const DatePickerBase = kind({
 		 */
 		dayAriaLabel: PropTypes.string,
 
+		/**
+		 * When it's `true` it changes the direction of the transition animation for day.
```
Comment (MikyungKim) [inline DatePicker/DatePickerBase.js:98]:
```suggestion
		 * When it's `true`, it changes the direction of the transition animation for the day.
```

**Comment by 0x64** [comment]
@MikyungKim Sorry, I forgot to answer. Let's merge this if it looks good to you.

**Pull request merged**